src/types.rs

```diff
@@ -26,4 +31,133 @@ pub enum Authority {
   ManagedNode,    // in our group and routing table
   ManagedClient,  // in our group
   Client,         // detached
+  Unknown
+}
+
+impl Encodable for Authority {
+  fn encode<E: Encoder>(&self, e: &mut E)->Result<(), E::Error> {
+    let mut authority = "";
+    match *self {
+      Authority::ClientManager => authority = "ClientManager",
+      Authority::NaeManager => authority = "NaeManager",
+      Authority::NodeManager => authority = "NodeManager",
+      Authority::ManagedNode => authority = "ManagedNode",
+      Authority::ManagedClient => authority = "ManagedClient",
+      Authority::Client => authority = "Client",
+      Authority::Unknown => authority = "Unknown",
+    }
+    CborTagEncode { tag : 5483_000 , data : &authority }.encode(e)
```

Comment:
agreed. we shall come up with a pattern for this tag.
say : 5483_ for madsafe, then 3 digits, the first for library (0 for types, 1 for routing, 2 for vault, etc.), last two for the number of parameters capsulated (or shall be for each datatype ?)